VirtualEconomyFramework/VENFTApp-Blazor/Components/Account/TransactionsList.razor

```diff
@@ -1,294 +1,122 @@
+                            <button class="btn btn-primary" @onclick="() => ShowTxDetails(context.Info.Txid)">
+                                Details
+                            </button>
+                        </VENFTApp_Blazor.Components.Column>
+                    </VENFTApp_Blazor.Components.Table>
+                    <VENFTApp_Blazor.Components.Pagination Total="@Transactions.Count" PerPage="@PerPage" OnChange="@HandlePaginationChanged" />
                 }
             </div>
         </div>
-    </div>
+    }
 </div>
 
-<Modal Title="Transaction Details"
-        Visible="@txDetailsVisible"
-        OnOk="@CloseTxDetails"
-        OnCancel="@CloseTxDetails">
-    <VENFTApp_Blazor.Components.Account.TransactionDetails TxId="@txDetailsTxId" SubAccount="@SubAccount" Address="@Address"/>
+<Modal Title="Transaction Details" Visible="@txDetailsVisible" OnOk="@CloseTxDetails" OnCancel="@CloseTxDetails">
+    <VENFTApp_Blazor.Components.Account.TransactionDetails TxId="@txDetailsTxId" SubAccount="@SubAccount" Address="@Address" />
 </Modal>
 
 @code {
-
-    [Parameter]
-    public int MaxLoadedTx
-    {
-        get => maxLoadedTxs;
-        set => maxLoadedTxs = value;
-    }
-    [Parameter]
-    public EventCallback<int> MaxLoadedTxChanged { get; set; }
-
     [Parameter]
     public bool SubAccount { get; set; } = false;
-
-    private string _address = string.Empty;
     [Parameter]
-    public string Address
-    {
-        get => _address;
-        set
-        {
-            if (!string.IsNullOrEmpty(value) && value != _address)
-            {
-                _address = value;
-                Load();
-            }
-        }
-    }
+    public string Address { get; set; } = null;
 
-    private bool txDetailsVisible = false;
-
-    private int maxLoadedTxs = 25;
+    private int PerPage { get; set; } = 10;
+    private int Offset { get; set; } = 0;
 
+    private bool txDetailsVisible = false;
     private string txDetailsTxId = string.Empty;
 
     private List<string> Transactions = new List<string>();
+    private List<TxDetails> TransactionsDetails = null;
 
-    private class TxDetails
+    protected override async Task OnInitializedAsync()
     {
-        public GetTransactionInfoResponse Info { get; set; } = new GetTransactionInfoResponse();
-        public string Sender { get; set; } = string.Empty;
-        public string Receiver { get; set; } = string.Empty;
-        public bool FromAnotherAccount { get; set; } = false;
-        public bool FromSubAccount { get; set; } = false;
-        public DateTime Time { get; set; } = DateTime.MinValue;
-    }
-    private List<TxDetails> TransactionsDetails = new List<TxDetails>();
-
-    protected override Task OnAfterRenderAsync(bool firstRender)
-    {
-        if (firstRender) Load();
-        return base.OnAfterRenderAsync(firstRender);
+        if (AppData.Account != null)
+            AppData.Account.Refreshed += RefreshedHandler;
+        await Load();
     }

```

Comment:
Component is automatically rendered after this

Reply:
ok